x-pack/plugins/data_visualizer/public/application/index_data_visualizer/components/index_data_visualizer_view/index_data_visualizer_view.tsx

```diff
@@ -112,57 +116,98 @@ export const getDefaultDataVisualizerListState = (): Required<DataVisualizerInde
+export interface IndexDataVisualizerViewProps {
+  currentIndexPattern: IndexPattern;
+  currentSavedSearch: SavedSearchSavedObject | null;
+}
+const restorableDefaults = getDefaultDataVisualizerListState();
+
+export const IndexDataVisualizerView: FC<IndexDataVisualizerViewProps> = (dataVisualizerProps) => {
   const {
-    services: { lens: lensPlugin, docLinks },
-  } = useMlKibana();
+    services: { lens: lensPlugin, docLinks, notifications, uiSettings },
+  } = useDataVisualizerKibana();
+  const { toasts } = notifications;
 
   const [dataVisualizerListState, setDataVisualizerListState] = usePageUrlState(
-    ML_PAGES.DATA_VISUALIZER_INDEX_VIEWER,
+    DATA_VISUALIZER_INDEX_VIEWER,
     restorableDefaults
   );
-  const [currentSavedSearch, setCurrentSavedSearch] = useState(mlContext.currentSavedSearch);
+  const [globalState, setGlobalState] = useUrlState('_g');
+
+  const [currentSavedSearch, setCurrentSavedSearch] = useState(
+    dataVisualizerProps.currentSavedSearch
+  );
+
+  const { currentIndexPattern } = dataVisualizerProps;
+
+  useEffect(() => {
+    if (dataVisualizerProps?.currentSavedSearch !== undefined) {
+      setCurrentSavedSearch(dataVisualizerProps?.currentSavedSearch);
+    }
+  }, [dataVisualizerProps?.currentSavedSearch]);
+
+  const getTimeBuckets = useCallback(() => {
+    return new TimeBuckets({
+      [UI_SETTINGS.HISTOGRAM_MAX_BARS]: uiSettings.get(UI_SETTINGS.HISTOGRAM_MAX_BARS),
+      [UI_SETTINGS.HISTOGRAM_BAR_TARGET]: uiSettings.get(UI_SETTINGS.HISTOGRAM_BAR_TARGET),
+      dateFormat: uiSettings.get('dateFormat'),
+      'dateFormat:scaled': uiSettings.get('dateFormat:scaled'),
+    });
+  }, [uiSettings]);
 
-  const { combinedQuery, currentIndexPattern, kibanaConfig } = mlContext;
   const timefilter = useTimefilter({
-    timeRangeSelector: currentIndexPattern.timeFieldName !== undefined,
+    timeRangeSelector: currentIndexPattern?.timeFieldName !== undefined,
     autoRefreshSelector: true,
   });
 
-  const { toasts } = useNotifications();
   const dataLoader = useMemo(() => new DataLoader(currentIndexPattern, toasts), [
     currentIndexPattern,
+    toasts,
   ]);
 
-  const [globalState, setGlobalState] = useUrlState('_g');
   useEffect(() => {
     if (globalState?.time !== undefined) {
       timefilter.setTime({
         from: globalState.time.from,
         to: globalState.time.to,
       });
     }
-  }, [globalState?.time?.from, globalState?.time?.to]);
+  }, [globalState, timefilter]);
+
   useEffect(() => {
     if (globalState?.refreshInterval !== undefined) {
       timefilter.setRefreshInterval(globalState.refreshInterval);
     }
-  }, [globalState?.refreshInterval?.pause, globalState?.refreshInterval?.value]);
+  }, [globalState, timefilter]);
 
   const [lastRefresh, setLastRefresh] = useState(0);
 
   useEffect(() => {
-    timeBasedIndexCheck(currentIndexPattern, true);
-  }, []);
+    if (!currentIndexPattern.isTimeBased()) {
```

Comment:
Note that this check and warning probably only apply when using the data visualizer inside of ML.

Reply:
Added this item to the meta issue #101435 👍